src/ray/core_worker/transport/direct_task_transport.h

```diff
@@ -178,31 +179,52 @@ class CoreWorkerDirectTaskSubmitter {
   /// (1) The lease client through which the worker should be returned
   /// (2) The expiration time of a worker's lease.
   /// (3) The number of tasks that are currently in flight to the worker
+  /// (4) The resources assigned to the worker
+  /// (5) The SchedulingKey assigned to tasks that will be sent to the worker
   struct LeaseEntry {
     std::shared_ptr<WorkerLeaseInterface> lease_client_;
     int64_t lease_expiration_time_;
     uint32_t tasks_in_flight_;
-
-    LeaseEntry(std::shared_ptr<WorkerLeaseInterface> lease_client = nullptr,
-               int64_t lease_expiration_time = 0, uint32_t tasks_in_flight = 0)
+    google::protobuf::RepeatedPtrField<rpc::ResourceMapEntry> assigned_resources_;
+    SchedulingKey scheduling_key_;
+
+    LeaseEntry(
+        std::shared_ptr<WorkerLeaseInterface> lease_client = nullptr,
+        int64_t lease_expiration_time = 0, uint32_t tasks_in_flight = 0,
+        google::protobuf::RepeatedPtrField<rpc::ResourceMapEntry> assigned_resources =
+            google::protobuf::RepeatedPtrField<rpc::ResourceMapEntry>(),
+        SchedulingKey scheduling_key = std::make_tuple(0, std::vector<ObjectID>(),
+                                                       ActorID::Nil()))
         : lease_client_(lease_client),
           lease_expiration_time_(lease_expiration_time),
-          tasks_in_flight_(tasks_in_flight) {}
+          tasks_in_flight_(tasks_in_flight),
+          assigned_resources_(assigned_resources),
+          scheduling_key_(scheduling_key) {}
   };
 
   // Map from worker address to a LeaseEntry struct containing the lease's metadata.
   absl::flat_hash_map<rpc::WorkerAddress, LeaseEntry> worker_to_lease_entry_
       GUARDED_BY(mu_);
 
-  // Keeps track of pending worker lease requests to the raylet.
-  absl::flat_hash_map<SchedulingKey,
-                      std::pair<std::shared_ptr<WorkerLeaseInterface>, TaskID>>
-      pending_lease_requests_ GUARDED_BY(mu_);
+  struct SchedulingKeyEntry {
+    // Keep track of pending worker lease requests to the raylet.
+    std::pair<std::shared_ptr<WorkerLeaseInterface>, TaskID> pending_lease_request_ =
```

Comment:
Could you remove the tail underscore at the end of these field names? Usually we do this only for private class members (see [Google style guide](https://google.github.io/styleguide/cppguide.html#Variable_Names)). Realizing now that the above `LeaseEntry` struct doesn't follow this convention, oops :)

Reply:
Should I also remove the underscores from the field names in the LeaseEntry struct? I guess it's better late than never, right?

Reply:
That would be great, thanks! :)